ci/gha/builds/macos-bazel.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
Can we have a:

```
# Usage
#    macos-bazel.sh [<target>...]
# 
# e.g.
#     macos-bazel.sh //google/cloud/... -//google/cloud/compute/...
```

Reply:
I added some comments, but these programs are not intended for humans to call directly, so I skipped the command-line validation